modules/EnsEMBL/Web/Utils/TrackHub.pm

```diff
@@ -204,6 +209,25 @@ sub get_hub {
   }
 }
 
+sub get_hub {
+### Fetch metadata about the hub and (optionally) its tracks
+### @param args Hashref (optional) 
+###                     - parse_tracks Boolean
+###                     - assembly_lookup Hashref
+### @return Hashref              
+###
+### This method calls get_hub_internal first with and, if there
+### is an error, then without. This means that caches shouldn't get "poisoned"
+### if they fail due to an intermittent failure on first population. 
+  my ($self, $args) = @_;
+
+  my $out = $self->get_hub_internal($args,1);
+  if($out->{'error'}) {
```

Comment:
What happen if $out is empty or broken somehow without 'error' field?

Reply:
Good point. I'll make this test stronger.